test/lib/Symlink_test.js

```diff
@@ -77,4 +78,248 @@ describe('Symlink', function() {
             this.symlink.resolve(this.connection);
         });
     });
+
+    describe('pre-hydration', function() {
+        describe('with simple prehydrated data', function() {
+            beforeEach(function() {
+                this.id = '/foo/123';
+                this.data = {
+                    id: 123,
+                    bar: 'test bar',
+                    derp: true
+                };
+                this.symlink = new Symlink(this.id, this.data);
+                this.connection = {
+                    get: sinon.stub()
+                        .withArgs('/foo/123')
+                        .returns(Promise.resolve({
+                            flerp: 'test flerp'
+                        }))
+                };
+            });
+
+            describe('all requested props prehydrated', function() {
+                beforeEach(function() {
+                    this.props = ['id', 'bar', 'derp'];
```

Comment:
can you aslo add a test with props that are more complex such as items@user.userId and check if it actually resolves with data passed in that's also an array?